account_statement_payment/models/statement_line.py

```diff
@@ -0,0 +1,105 @@
+# © 2022 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+ACTION = False
+SALE = False
+PURCHASE = False
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = "account.bank.statement.line"
+
+    origin_statement = fields.Char(compute="_compute_origin_statement")
+
+    def _compute_origin_statement(self):
+        global ACTION, SALE, PURCHASE
+        if not ACTION:
+            ACTION = self.sudo().env.ref(
+                "account.action_account_payments_payable", raise_if_not_found=False
+            )
+        base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
+        # Sale or purchase modules may be not both installed,
+        # we have to check outside of the loop and defined in a global scope
+        if "sale.order" in self.env.registry.models.keys():
+            SALE = "sale"
+        if "purchase.order" in self.env.registry.models.keys():
+            PURCHASE = "purchase"
+        for rec in self:
+            rec_lines = rec.move_id.line_ids.filtered(lambda s: s.full_reconcile_id)
+            line_rec = {
+                x: x.full_reconcile_id
+                for x in rec.move_id.line_ids.filtered(lambda s: s.full_reconcile_id)
+            }
+            reconc = self.env["account.move.line"].search(
+                [
+                    (
+                        "full_reconcile_id",
+                        "in",
+                        rec_lines.mapped("full_reconcile_id").ids,
+                    ),
+                ]
+            )
+            pay = {
+                x.full_reconcile_id: x.mapped("payment_id")
+                for x in reconc.filtered(lambda s: s.payment_id)
+            }
+            line_docs = {
+                line: "%s, %s %s %s %s"
+                % (
+                    pay[re].amount,
+                    pay[re].ref or pay[re].name,
```

Comment:
@bealdav `pay[re].name or pay[re].ref `
because payment's ref is usually the PO's name, so you have 2 times the PO name.

Reply:
I will remove the old way to compute